### PR TITLE
Fix various usability bugs

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
This PR fixes a range of usability bugs:

1. Log an error instead of exiting on zenml version downgrades and point to the `zenml clean` command as an alternative to fixing potential schema validation errors
2. Don't fail `zenml service start/stop` if service schema changes
3. Highlight logged errors and warnings using the rich color schema
4. Ensure a model server is always running in the standard mlflow deployer step
5. Don't trim UUID in `zenml served-models list`
6. Ensure a model server is always running in the seldon deployer step
7. Improved Kubeflow orchestrator validations:

 * fail if the configured `kubernetes_context` is not locally available
 * warn if the default or configured `kubernetes_context` is not the same as the currently active kube context
 * improved error messages for validations involving local component flavors to show the configured (or default) `kubernetes_context` attribute and point out some possible corrective actions

8. Allow more flexibility in the Kubeflow orchestrator configurations, to allow for use-cases such as using the Kubeflow orchestrator with a user-managed k3d cluster or with a user-provided way of sharing the local folders with an external Kubernetes cluster:
 * add a `skip_local_validations` attribute that can be set to bypass the local validations
 * add `skip_cluster_provisioning` and `skip_ui_daemon_provisioning` attributes that can be used to skip the provisioning of the local cluster and KFP UI daemon

9.  Make stack and component name optional in update cli commands
10. Better logging messages and cleanup for failed local daemons

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/mlflow/steps/mlflow_deployer.py [hunk 1]
```diff
@@ -134,25 +132,48 @@ def mlflow_model_deployer_step(
...
-            raise RuntimeError(
+            logger.warning(
                 f"An MLflow model with name `{config.model_name}` was not "
-                f"trained in the current pipeline run and no previous "
-                f"service was found."
+                f"logged in the current pipeline run and no running MLflow "
+                f"model server was found. Please ensure that your pipeline "
+                f"includes an `mlflow_enable` decorated step that trains a "
+                f"model and logs it to MLflow. This could also happen if "
+                f"re-training a model in the current pipeline run was skipped "
```
Comment (strickvl) [inline src/zenml/integrations/mlflow/steps/mlflow_deployer.py:145]: Is there something missing between 'if' and 're-training'? Maybe it's missing a 'while'?

File: src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py [hunk 2]
```diff
@@ -654,7 +734,7 @@ def resume(self) -> None:
                 kubernetes_context=kubernetes_context
             )
 
-        if not self.is_daemon_running:
+        if not self.skip_ui_daemon_provisioning and not self.is_daemon_running:
```
Comment (schustmi) [inline src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py:737]: Do we need this explicit check for `self.skip_ui_daemon_provisioning` here? I guess it's already covered in `self.is_daemon_running` right? (Same thing on the next change below)

File: src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py [hunk 3]
```diff
@@ -413,9 +475,10 @@ def _upload_and_run_pipeline(
                     )
         except urllib3.exceptions.HTTPError as error:
             logger.warning(
-                "Failed to upload Kubeflow pipeline: %s. "
-                "Please make sure your kube config is configured and the "
-                "current context is set correctly.",
+                f"Failed to upload Kubeflow pipeline: %s. "
+                f"Please make sure your kube config is configured and the "
+                f"{self.kubernetes_context} kubernetes context is set "
```
Comment (schustmi) [inline src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py:480]: I think `set correctly` isn't really fitting here anymore, maybe configured?

File: src/zenml/integrations/mlflow/steps/mlflow_deployer.py [hunk 4]
```diff
@@ -86,23 +86,21 @@ def mlflow_model_deployer_step(
     """
     model_deployer = MLFlowModelDeployer.get_active_model_deployer()
 
-    if not config.model_uri:
-        # fetch the MLflow artifacts logged during the pipeline run
-        experiment_tracker = Repository(  # type: ignore[call-arg]
-            skip_repository_check=True
-        ).active_stack.experiment_tracker
+    # fetch the MLflow artifacts logged during the pipeline run
+    experiment_tracker = Repository(  # type: ignore[call-arg]
```
Comment (schustmi) [inline src/zenml/integrations/mlflow/steps/mlflow_deployer.py:90]: Not too familiar with the entire deployment stuff, but it seems like (at least in this method) `config.model_uri` is not used anymore. Can we get rid of it or does it serve any purpose in different parts of the code?

File: src/zenml/integrations/mlflow/steps/mlflow_deployer.py [hunk 5]
```diff
@@ -134,25 +132,48 @@ def mlflow_model_deployer_step(
...
+        # the currently running service created for the same model, if any
         if not existing_services:
-            raise RuntimeError(
+            logger.warning(
                 f"An MLflow model with name `{config.model_name}` was not "
-                f"trained in the current pipeline run and no previous "
-                f"service was found."
+                f"logged in the current pipeline run and no running MLflow "
+                f"model server was found. Please ensure that your pipeline "
+                f"includes an `mlflow_enable` decorated step that trains a "
```
Comment (schustmi) [inline src/zenml/integrations/mlflow/steps/mlflow_deployer.py:143]:
```suggestion
                f"includes an `@enable_mlflow` decorated step that trains a "
```

**Review by schustmi — CHANGES_REQUESTED** [review]
It seems like you're improving the kubeflow error messages daily, I love it!

File: src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py (hunk 2 above)
Comment (stefannica) [reply src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py:737]: True, we don't need it here, but we do need it below, because the daemon is always reported as running if `self.skip_ui_daemon_provisioning`

File: src/zenml/integrations/mlflow/steps/mlflow_deployer.py (hunk 4 above)
Comment (stefannica) [reply src/zenml/integrations/mlflow/steps/mlflow_deployer.py:90]: `config.model_uri` may not be used here anymore, but it still needs to be part of the service configuration. It's just set elsewhere (in the model deployer).

File: src/zenml/integrations/mlflow/steps/mlflow_deployer.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/integrations/mlflow/steps/mlflow_deployer.py:145]: you're right, it reads awkwardly, I'll rephrase it to be easier to read

**Review by schustmi — APPROVED** [review]

**Review by htahir1 — APPROVED** [review]
Beautiful PR!
